pandas/core/generic.py

```diff
@@ -11772,3 +11774,12 @@ def _doc_params(cls):
     The required number of valid values to perform the operation. If fewer than
     ``min_count`` non-NA values are present the result will be NA.
 """
+
+
+def validate_ascending(ascending):
```

Comment:
can you move to pandas/util/validators.py

shouldn't this raise for a list-like case if something is wrong?

Reply:
e.g. this shouldn't return anything

pls add some typing

Reply:
It raises for a list-like case if any of the items in ``ascending`` list is not treated as a valid boolean.
This construct does just that.
```
return [validate_bool_kwarg(item, "ascending", **kwargs) for item in ascending]
```

Reply:
this can simply be part of validate_bool_kwarg or another function but should be co-located with validate_bool_kwarg

Reply:
Moved the function to ``pandas/util/_validators.py``.
It is not clear for me at the moment how to make this validation anyhow different (as a part of ``validate_bool_kwarg``) due to the fact that multiple types are acceptable for ``ascending``.